src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/OrganizationMapperTest.java

```diff
@@ -1,36 +1,139 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.Coding;
 import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.codesystems.OrganizationType;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.InvolvedParty;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
 
-import java.util.ArrayList;
+@ExtendWith(MockitoExtension.class)
+class OrganizationMapperTest {
 
-import static org.assertj.core.api.Assertions.assertThat;
+    @Mock
+    private Message message;
 
-class OrganizationMapperTest {
+    @Mock
+    private UUIDGenerator uuidGenerator;
+
+    @InjectMocks
+    private OrganizationMapper mapper;
 
     @Test
-    void testMapPerformingOrganization() {
-        final Message message = new Message(new ArrayList<>());
+    void testMapMessageToRequestingOrganizationWithNoRequesterDetails() {
+        when(message.getInvolvedParties()).thenReturn(Collections.emptyList());
 
-        assertThat(new OrganizationMapper().mapToPerformingOrganization(message))
-            .isExactlyInstanceOf(Organization.class);
+        assertThat(mapper.mapToRequestingOrganization(message)).isEmpty();
     }
 
     @Test
-    void testMapRequestingOrganization() {
-        final Message message = new Message(new ArrayList<>());
+    void testMapMessageToRequestingOrganization() {
+        final var requestingParty = new InvolvedParty(List.of(
+            "ignore me",
+            "NAD+PO+++MATTHEW?'S GP",
+            "SPR+ORG",
+            "ignore me"
+        ));
 
-        assertThat(new OrganizationMapper().mapToRequestingOrganization(message))
-            .isExactlyInstanceOf(Organization.class);
+        when(message.getInvolvedParties()).thenReturn(List.of(requestingParty));
+
+        Optional<Organization> result = mapper.mapToRequestingOrganization(message);
+        assertThat(result).isNotEmpty();
+
+        Organization organization = result.get();
+
+        assertAll(
+            () -> assertThat(organization.getName()).isEqualTo("MATTHEW'S GP"),
+            () -> assertThat(organization.getType()).isEmpty()
+        );
```

Comment:
Can be chained if you want:
```suggestion
        Optional<Organization> result = mapper.mapToRequestingOrganization(message);
        assertThat(result).isNotEmpty()
            .hasValueSatisfying(() -> assertAll(
                () -> assertThat(organization.getName()).isEqualTo("MATTHEW'S GP"),
                () -> assertThat(organization.getType()).isEmpty()
            ));
```

Reply:
Similarly in other tests here

Reply:
Made this change in most of the tests. Causing issues in the `testMapMessageToPerformingOrganizationWithOrganizationAndDepartment` test so I've left that out

<img width="1017" alt="Screenshot 2021-03-04 at 16 39 49" src="https://user-images.githubusercontent.com/7227831/109998104-d0ba1b00-7d08-11eb-9e65-d8de3dd63203.png">